swift/ql/src/queries/Security/CWE-916/InsufficientHashIterations.swift

```diff
@@ -0,0 +1,14 @@
+
+func encrypt() {
+	// ...
+
+	// BAD: Using insufficient (i.e., < 120,000) hash iterations keys for encryption
+	_ = try PKCS5.PBKDF1(password: getRandomArray(), salt: getRandomArray(), iterations: 90000, keyLength: 0)
+	_ = try PKCS5.PBKDF2(password: getRandomArray(), salt: getRandomArray(), iterations: 90000, keyLength: 0)
+
+	// GOOD: Using sufficient (i.e., >= 120,000) hash iterations keys for encryption
+	_ = try PKCS5.PBKDF1(password: getRandomArray(), salt: getRandomArray(), iterations: 120120, keyLength: 0)
+	_ = try PKCS5.PBKDF2(password: getRandomArray(), salt: getRandomArray(), iterations: 120120, keyLength: 0)
```

Comment:
Judging by the OWASP reference, what is an adequate number of iterations seems to vary according to the algorithm you're using. It's not entirely clear to me looking at this example, whether or not this is adequate for the algorithm because I don't see which hashing algorithm is used.

This may be entirely unimportant - I'm a writer, not a developer 🙂

Reply:
I think this is a good point.  We don't _implement_ the different recommended values for different algorithms, rather we just took the lowest recommendation and check the value used is at least that.  In practice this design is likely to catch the vast majority of problems and keeps the query simple and reliable - but there's no reason our _examples_ shouldn't set a good example by being fully OWASP compliant.

Reply:
Agreed. I've updated the test to be fully OWASP compliant :)